Improve Content fields documentations

MikeAlhayek opened the pull request:
<!--- Please make sure that you're familiar with our contribution guidelines before submitting a pull request: https://docs.orchardcore.net/en/latest/contributing/. -->

Piedone commented:
If you did this with the help of AI, did you check the whole text, including that there's no important information now removed, neither useless fluff added?

MikeAlhayek commented:
> If you did this with the help of AI, did you check the whole text, including that there's no important information now removed, neither useless fluff added?

Yes. Everything seems there. I added the recipe at the bottom and the available editors for each field

Piedone commented on src/docs/reference/modules/ContentFields/README.md:
```diff
@@ -189,40 +154,27 @@ per set based on the request culture, if no culture is specified.
-### `UserPicker Field`
-
-The User Picker field allows you to relate users to a content item.
-
-When adding the field to a content type, use the settings to specify whether to
+### `UserPickerField`
 
-- List all users,
-- List users from specific roles.
```
The new text contains just a fraction of the information here.

Piedone commented on src/docs/reference/modules/ContentFields/README.md:
````diff
@@ -231,66 +183,36 @@ When adding the field to a content type, use the settings to specify whether to
 === "Razor"
-
-    ``` html
-    @model OrchardCore.ContentFields.ViewModels.DisplayUserPickerFieldViewModel
-    @using OrchardCore.Mvc.Utilities
````
Without this the code doesn't work.

Piedone commented on src/docs/reference/modules/ContentFields/README.md:
```diff
@@ -231,66 +183,36 @@ When adding the field to a content type, use the settings to specify whether to
-be used to edit a field differently, for instance the WYSIWYG HTML editor is a custom editor for the HTML
-field. Formatters are alternate shapes that can be used to render a field on the front end, for instance
-a `Link` field could be rendered as a Youtube video player.
+Consider creating custom editors/formatters before new fields. Fields should represent specific data types, while editors/formatters handle input/output variations.
```
Formatters are not explained anymore. There's a bunch of omissions from this downwards too, but I won't point them out one by one, please just check.

Piedone reviewed (changes requested):
It doesn't seem so to me that everything is there. Making the documentation more succinct and removing duplications is great. But we shouldn't remove actual information.

MikeAlhayek commented on src/docs/reference/modules/ContentFields/README.md:
````diff
@@ -1,186 +1,151 @@
     }
     ```
 
-Or to render the referenced content item:
-
-=== "Liquid"
-
-    ``` liquid
-    {% assign contentItems = Model.ContentItemIds | content_item_id %}
-    {% for contentItem in contentItems %}
-        {{ contentItem | shape_build_display: "Detail" | shape_render }}
-    {% endfor %}
-    ```
-
-=== "Razor"
-
-    ``` html
-    @foreach (var contentItem in await Orchard.GetContentItemsByIdAsync(Model.ContentItemIds))
-    {
-        @await Orchard.DisplayAsync(contentItem, "Detail")
-    }
-    ```
````
it's added

Piedone commented on src/docs/reference/modules/ContentFields/README.md:
```diff
@@ -1,33 +1,46 @@
+!!! note
+    While the UI allows adding fields directly to content types, behind the scenes it creates a content part with the same name as the content type and attaches the fields to it.
+
+Fields support multiple editors - different UIs for data input. For example, a TextField can have editors like `Email`, `TextArea`, or `Tel`. Editors can be configured in the field settings.
```
```suggestion
Fields support multiple editors — different UIs for data input. For example, a TextField can have editors like `Email`, `TextArea`, or `Tel`. Editors can be configured in the field settings.
```

MikeAlhayek replied:
Can't tell the difference

Piedone replied:
It's an em dash: https://www.merriam-webster.com/grammar/em-dash-en-dash-how-to-use. What you used the hyphen for is covered by that. Punctuation nitpick.

Piedone commented on src/docs/reference/modules/ContentFields/README.md:
```diff
@@ -1,33 +1,46 @@
-| `TimeField`                         | `TimeSpan? Value`                                              |
-| `UserPickerField`                   | `string[] UserIds`                                             |
-| `YoutubeField`                      | `string EmbeddedAddress, string RawAddress`                    |
+| Name                                | Properties                                                     | Available Editors |
```
The formatting is off here. I guess all of the 

```
Standard ``
```

were meant to be

```
`Standard`
```
?

MikeAlhayek replied:
The default editor is named "Standard" and the value for the Standard is empty string. So (Standard ``) shows the name and the corresponding value for each. So I think it's correct

Piedone replied:
Then something like "Standard (empty)" would be clearer anyway. But the main problem here is that due to the backticks the code formatting is broken (check out by running the docs site locally).

MikeAlhayek replied:
@Piedone done.
